File: php/features.php
<?php
declare(strict_types=1);

// Launchpics SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LaunchpicsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LaunchpicsBaseFeature();
            case "test":
                return new LaunchpicsTestFeature();
            default:
                return new LaunchpicsBaseFeature();
        }
    }
}

File: py/launchpics_sdk/features.py
# Launchpics SDK feature factory

from launchpics_sdk.feature.base_feature import LaunchpicsBaseFeature
from launchpics_sdk.feature.test_feature import LaunchpicsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LaunchpicsBaseFeature(),
        "test": lambda: LaunchpicsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
